channeldb/peers_test.go

```diff
@@ -0,0 +1,41 @@
+package channeldb
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/routing/route"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestFlapCount tests lookup and writing of flap count to disk.
+func TestFlapCount(t *testing.T) {
+	db, cleanup, err := makeTestDB()
+	require.NoError(t, err)
+	defer cleanup()
+
+	// Try to read flap count for a peer that we have no records for.
+	_, err = db.ReadFlapCount(testPub, testChanPoint1)
+	require.Equal(t, ErrNoPeerBucket, err)
+
+	var flapCount uint32 = 20
+	peers := map[route.Vertex]map[wire.OutPoint]uint32{
+		testPub: {
```

Comment:
Test could have at least two peers and at least one peer could have more than one channel to make sure serialization is correct.